File: petclinic-backend/src/main/java/victor/training/petclinic/rest/dto/PetTypeFieldsDto.java
package victor.training.petclinic.rest.dto;

import jakarta.validation.constraints.*;
import io.swagger.v3.oas.annotations.media.Schema;

public class PetTypeFieldsDto {
    @NotNull
    @Size(min = 1, max = 80)
    @Schema(example = "cat", description = "The name of the pet type.")
    private String name;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: petclinic-backend-ts/src/pet-types/dto/pet-type-fields.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsDefined, Length } from 'class-validator';

/**
 * Ported from victor.training.petclinic.rest.dto.PetTypeFieldsDto.
 */
export class PetTypeFieldsDto {
  @IsDefined({ message: 'must not be null' })
  @Length(1, 80, { message: 'size must be between 1 and 80' })
  @ApiProperty({ example: 'cat', description: 'The name of the pet type.' })
  name!: string;
}
